### ISSUE TITLE
How to move from standard pdf.js to flipbook?

### ISSUE THREAD

**Issue by jvieille** [iberan/pdfjs-flipbook#1]
I am using pdf.js from Drupal, how can I make use of this flipbook enhancement?
A mere replacement of my current prebuilt pdf.js version dated 14/08/2017 does not work.
Is this project alive ?
Any workable solution for flipping pages with pdf.js?
Thanks

**Comment by iberan** [comment]
I don’t have any experience in Drupal, sorry. Here you are a standalone workable solution by pdfjs-flipbook:

http://www.tanchaz.hu/work/pdfjs-flipbook/web/viewer.html?file=newspaper.pdf

It works in normal, magazine (flipbook) or slide show mode as well.

**Comment by jvieille** [comment]
I managed to get it working.
- added the turnjs library, magazine.js and magazine.css
- completed local terms
- completed images
- modified viewer.html viewer.css, and viewer.js

That's great!!!
You an see it in action here https://www.see.asso.fr/node/23113/landing by clicking on the lens "voir l'article"
How to default on magazine mode ?
Why is that functionality was never included in pdf.js project?

Thanks a lot!!!

**Comment by iberan** [comment]
Clear the browser cash (!!!) and try it:

http://www.tanchaz.hu/work/pdfjs-flipbook/web/viewer.html?file=newspaper.pdf#magazineMode=true

I hope, it helps anyway for you and others.

**Comment by jvieille** [comment]
Yes it helps! I understand that I just need to add #magazineMode=true to the source file url.
Thanks !!!

**Comment by jvieille** [comment]
By the , what I did was just migrating your code to the latest pdf.js pre-built https://github.com/mozilla/pdf.js/releases/download/v1.9.426/pdfjs-1.9.426-dist.zip

**Comment by iberan** [comment]
- You have the new **dist** directory wich contains the **build** and **web** directory.
Let's copy there the **turnjs4** directory from **pdfjs-flipbook**.

- Copy into **dist/web** directory these files from **pdfjs-flipbook/web**:
 _magazine.css_
_l10n.js_
_magazine.js_

- and copy into **dist/web/images** from **pdfjs-flipbook/images**:
 _toolbarButton-Exit.png_
_toolbarButton-magazineMode.png_

- Open _dist/web/viewer.html_ and insert between `<head></head>` tags:

`<link rel="stylesheet" href="magazine.css" />`
- and insert above the `</body>` tag:
```
<!-- jquery may not necessary in cms -->
<script src="https://code.jquery.com/jquery-2.1.4.min.js"></script>
<!-- turn.min.js, zoom.min.js, magazine.js -->
<script src="l10n.js"></script>
<script src="../turnjs4/lib/turn.min.js"></script>
<script src="../turnjs4/lib/zoom.min.js"></script>
<script src="debugger.js"></script>
<script src="magazine.js"></script>
<script type="text/javascript">
$(document).ready(function () {
MagazineView.init();
});
</script>
```
- Open _dist/web/viewer.css_ and insert:

`.toolbarButton.magazineMode::before,
.secondaryToolbarButton.magazineMode::before {
content: url(images/toolbarButton-magazineMode.png);
}
`

That's all.
